src/components/Provider.tsx

```diff
@@ -19,14 +22,15 @@ export interface ProviderProps<A extends Action = AnyAction> {
   children: ReactNode
 }
 
-function Provider({ store, context, children }: ProviderProps) {
-  const contextValue = useMemo(() => {
+function Provider({ store, context, children, serverState }: ProviderProps) {
+  const contextValue: ReactReduxContextValue = useMemo(() => {
     const subscription = createSubscription(store)
     return {
       store,
       subscription,
+      getServerState: serverState ? () => serverState : undefined,
     }
-  }, [store])
+  }, [store, serverState])
```

Comment:
If I understand correctly the `serverState` should be an immutable value, used only for hydration purposes. Might be worth skipping this here? Or putting it in a ref (that would be read by `getServerState`) to make this more explicit? Obviously, that would break the rules of hooks but IMHO it's sometimes worth it to represent some constraints etc